### PR TITLE
Make Value Def Generic 

### PR DESCRIPTION
**Pull request by kanitw**
so schema is more accurate

Depends on #4426 and #4430.

### CODE REVIEW

File: scripts/rename-schema.sh [hunk 1]
```diff
@@ -11,6 +11,14 @@ perl -pi -e s,'Generic(.*)<FacetedUnitSpec\,LayerSpec>','\1',g build/vega-lite-s
 perl -pi -e s,'GenericUnitSpec<EncodingWithFacet\,AnyMark>','FacetedCompositeUnitSpecAlias',g build/vega-lite-schema.json
 perl -pi -e s,'GenericUnitSpec<Encoding\,AnyMark>','CompositeUnitSpecAlias',g build/vega-lite-schema.json
 
+perl -pi -e s,'FieldDefWithoutScale','FieldDef',g build/vega-lite-schema.json
```
Comment (domoritz) [inline scripts/rename-schema.sh]: Do you think that we might ever accidentally rename two distinct properties that then become the same? Any idea how we could test it automatically?

Comment (kanitw) [reply scripts/rename-schema.sh]: If that's the case, we will have redundant keys in `schema.definitions` and the schema will no longer be a valid json.

**Comment by domoritz** [comment]
still wip?

**Comment by kanitw** [comment]
> still wip?

Yes -- look a the schema diff.  still terrible.

**Comment by domoritz** [comment]
@kanitw What do you think about this PR? If we want it, we should try to get it in before 3.0 to not break the schema later.

**Pull request merged**